File: tmp/test_stft_many.py
# coding: utf-8

import numpy as np
import tools.stft as STFT
from scipy.signal import get_window


N = 8*6
NFFT = 8
WINDOW = 8
NOVERLAP = 0
HOP = NFFT-NOVERLAP
ZERO_PHASE_WINDOWING = 1


def main():
    x = np.array(range(N))
    w = get_window('hanning', WINDOW, False)
    mX, pX = STFT.stftAnal(x, w, NFFT, HOP,
                           zero_phase_windowing=ZERO_PHASE_WINDOWING)
    print 'mX: {}'.format(mX)
    print 'pX: {}'.format(pX)


if __name__ == '__main__':
    main()

File: tmp/test_stft_many.m
N = 8*6;
NFFT = 8;
Fs = 8000; % used for scaling plots
WINDOW = NFFT;
NOVERLAP = 0;
HOP = NFFT - NOVERLAP;
ZERO_PHASE_WINDOWING = 1;

x = 0:N-1;

w = vertcat([0], hanning(WINDOW-2), [0]);
w = w ./ sum(w);

y = xxspecgram(x, NFFT, Fs, w, NOVERLAP, ZERO_PHASE_WINDOWING);

mX = 20 * log10(abs(y))'
pX = unwrap(angle(y))'
